syntax = "proto3";
import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";


service PostService{
    rpc CreatePost (NewPost) returns (ResponseNewPost);
    rpc UpdatePost (RequestUpdatePost) returns (google.protobuf.Empty);
    rpc DeletePost (RequestDeletePost) returns (google.protobuf.Empty);
    rpc GetPostById (RequestGetPostById) returns (PostItem);
    rpc GetPosts (RequestGetPosts) returns (stream PostItem);
}

message NewPost {
  int64 user_id = 1;
  string title = 2;
  string content = 3;
}

message ResponseNewPost {
  int64 post_id = 1;
}

message RequestUpdatePost {
  int64 user_id = 1;
  int64 post_id = 2;
  string new_title = 3;
  string new_content = 4;
}

message RequestDeletePost {
  int64 user_id = 1;
  int64 post_id = 2;
}

message RequestGetPostById {
  int64 user_id = 1;
  int64 post_id = 2;
}

message RequestGetPosts {
  int64 user_id = 1;
}

message PostItem {
  int64 id = 1;
  int64 user_id = 2;
  google.protobuf.Timestamp time = 3;
  string title = 4;
  string content = 5;

}